src/regionprops.jl

```diff
@@ -180,16 +180,60 @@ function getbboxcolumns(props::DataFrame)
     return filter(col -> occursin(r"^bbox-\d$", col), names(props))
 end
 
+
+FloeLabelsImage = Union{BitMatrix, Matrix{<:Bool}, Matrix{<:Integer}}
+
 """
     cropfloe(floesimg, props, i)
 
 Crops the floe delimited by the bounding box data in `props` at index `i` from the floe image `floesimg`.
+
+If the dataframe has bounding box data `min_row`, `min_col`, `max_row`, `max_col`, but no `label`, then returns the largest contiguous component.
+
+If the dataframe has bounding box data `min_row`, `min_col`, `max_row`, `max_col`, and a `label`, then returns the component with the label. In this case, `floesimg` must be an Array{Int}.
+
+If the dataframe has only a `label` and no bounding box data, then returns the component with the label, padded by one cell of zeroes on all sides. In this case, `floesimg` must be an Array{Int}.
+
+
 """
-function cropfloe(floesimg::BitMatrix, props::DataFrame, i::Int64)
+function cropfloe(floesimg::FloeLabelsImage, props::DataFrame, i::Integer)
+    props_row = props[i, :]
+    colnames = names(props_row)
+    if "min_row" in colnames && "min_col" in colnames && "max_row" in colnames && "max_col" in colnames
+        if "label" in colnames
+            return cropfloe(
+                floesimg,
+                props_row.min_row,
+                props_row.min_col,
+                props_row.max_row,
+                props_row.max_col,
+                props_row.label
+            )
+        else
+            floesimg_bitmatrix = floesimg .> 0
+            return cropfloe(
+                floesimg_bitmatrix,
+                props_row.min_row,
+                props_row.min_col,
+                props_row.max_row,
+                props_row.max_col
+            )
+        end
+    elseif "label" in colnames
+        return cropfloe(floesimg, props_row.label)
+    end
```

Comment:
Consider a flatter approach:
```suggestion
    colnames = Set(names(props_row)
    bbox = ["min_row", "min_col", "max_row", "max_col"]
    bbox_label = [bbox..., "label"]

    issubset(Set(bbox_label), colnames) && return cropfloe(
        floesimg,
        props_row[bbox_label]...
    )

    issubset(Set(bbox), colnames) && return cropfloe(
        floesimg .> 0,
        props_row[bbox]...
    )

    "label" in colnames && return cropfloe(floesimg, props_row.label)
```

Reply:
I like the idea, and I've introduced the sets. 

I'd prefer to not have the list of column names doing double duty as both the set of names to check and the ordering of the columns for the output – that could lead to some really weird errors in future if someone who doesn't know that the order of the column names in the list is important and decides to (say) put them in alphabetical order. Having the order of the function arguments encoded in a list would also make it more difficult for refactoring software to help us refactor the functions if we need to in the future. So I think it's best to write those out explicitly. The performance loss will be minimal if anything, and I think this makes the code more explicit.

It would be a different story if we could use keyword arguments here, but that's not possible for the multiple-dispatch kind of function we're using.

Reply:
I like the idea, and I've introduced the sets. 

I'd prefer to not have the list of column names doing double duty as both the set of names to check and the ordering of the columns for the output – that could lead to some really weird errors in future if someone who doesn't know that the order of the column names in the list is important and decides to (say) put them in alphabetical order. Having the order of the function arguments encoded in a list would also make it more difficult for refactoring software to help us refactor the functions if we need to in the future. So I think it's best to write those out explicitly. The performance loss will be minimal if anything, and I think this makes the code more explicit.

It would be a different story if we could use keyword arguments here, but that's not possible for the multiple-dispatch kind of function we're using.